### DISCUSSION TITLE
Create a lifecycle diagram

### DISCUSSION THREAD

**Discussion by henk23** [nuxt/nuxt#16383] [Ideas]
When working with Nuxt 2 I always found this page super helpful: https://nuxtjs.org/docs/concepts/nuxt-lifecycle/

Can we make one of these graphics for Nuxt 3? I'm willing to contribute, but so far have too little insight into how the lifecycle actually works.

Some ideas on what should be included:

- What happens at build time
- What happens at server start
  - including Nitro plugins, middleware and (api) routes
- What happens between request and response on the server, in what order, separated into it's parts:
  - Nitro
  - Nuxt
  - Vue
- What happens on the client right after initial server response (hydration?)
- What happens on the client side after that
  - for example navigating to other routes, route middlewares, fetching async data

**Comment by arb85** [comment]
I'm also willing to contribute to this if needed

**Comment by Mini-ghost** [comment]
Hi! I’ve written some documentation about the Nuxt Lifecycle, and I hope it will be helpful to you. If you have any thoughts or suggestions about this document, please feel free to share them in this PR. I would greatly appreciate your feedback. Thank you!

👉 https://github.com/nuxt/nuxt/pull/30726
